### DISCUSSION TITLE
Error: ModuleNotFoundError: No module named 'PyQt5'

### DISCUSSION THREAD

**Discussion by ykzhu2019** [idaholab/moose#22858] [Q&A Getting Started] [answered]
Error popped out when I was trying to run peacock in the example ex01_inputfile in moose directory. Error:  

Error: ModuleNotFoundError: No module named 'PyQt5' 

I tried to run "conda install -c anaconda pyqt" but it did not help...

Any suggestions would be appreciated. 

Thanks.

**Comment by GiudGiud** [comment, marked as answer]
Hello

what do you mean it did not help? Did it not install it? What does mamba list say for pyqt?

you can also have try conda install pyqt5

what does ‘which Python’ return?

Guillaume

**Reply by ykzhu2019** [reply]
I ran which python and conda install pyqt5. There is an issue 

"PackagesNotFoundError: The following packages are not available from current channels" . 

See below:


(moose) ykzhu@MacBook-Pro ~ % which Python
/Users/ykzhu/mambaforge3/envs/moose/bin/Python
(moose) ykzhu@MacBook-Pro ~ % 
(moose) ykzhu@MacBook-Pro ~ % conda install pyqt5                       
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - pyqt5

Current channels:

  - https://conda.software.inl.gov/public/osx-arm64
  - https://conda.software.inl.gov/public/noarch
  - https://conda.anaconda.org/conda-forge/osx-arm64
  - https://conda.anaconda.org/conda-forge/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.


(moose) ykzhu@MacBook-Pro ~ %

**Reply by GiudGiud** [reply]
Ok I just checked and indeed you wont be able to run peacock in the same conda environment as moose. 

You need to create new one and install the moose-peacock package there
See this answer for similar conclusions
https://github.com/idaholab/moose/discussions/22718#discussioncomment-4264368
And this answer for the solution
https://github.com/idaholab/moose/discussions/21606#discussioncomment-3174230

**Reply by ykzhu2019** [reply]
Thanks. I will try that.
